src/main/scala/com/databricks/labs/overwatch/utils/DataFrameSyntax.scala

```diff
@@ -0,0 +1,69 @@
+package com.databricks.labs.overwatch.utils
+
+import org.apache.log4j.{Level,Logger}
+import org.apache.spark.sql.Dataset
+import java.io.ByteArrayOutputStream
+
+trait DataFrameSyntax {
+
+  /**
+    * Implements extension methods for Spark `Dataset[T]`s (including
+    * `DataFrame`s) that mimic Spark's built-in `Dataset.show()`
+    * family of overloaded methods.
+    *
+    * Instead of producing output on the console, they return an
+    * `Iterator[String]` suitable for redirecting output to a logger,
+    * for example, like this:
+    *
+    * {{{
+    * debugDF
+    *   .showLines( debugDF.count.toInt, 0, true)
+    *   .foreach( logger.log( Level.INFO, _))
+    * }}}
+    *
+    * Produces log entries according to your logger configuration:
+    *
+    * {{{
+    * 24/05/04 00:18:16 INFO Silver: -RECORD 0---------------------------------
+    * 24/05/04 00:18:16 INFO Silver:  organization_id       | 2753962522174656
+    * 24/05/04 00:18:16 INFO Silver:  jobId                 | 903015066329560
+    * 24/05/04 00:18:16 INFO Silver:  fromMS                | 1709419775381
+    * . . .
+    * 24/05/04 00:18:16 INFO Silver: -RECORD 1---------------------------------
+    * . . .
+    * }}}
+    */
+
+  implicit class DataFrameShower[T]( val df: Dataset[T]) {
+
+    def showLines(): Iterator[String] =
+      showLines( 20)
+
+    def showLines( numRows: Int): Iterator[String] =
+      showLines( numRows, truncate = true)
+
+    def showLines( truncate: Boolean): Iterator[String] =
+      showLines( 20, truncate)
+
+    def showLines( numRows: Int, truncate: Boolean): Iterator[String] =
+      if( truncate) {
+        showLines( numRows, truncate = 20)
+      } else {
```

Comment:
@neilbest-db  why 20 count here?

Reply:
Like the `truncate` argument for [native](https://spark.apache.org/docs/3.3.0/api/scala/org/apache/spark/sql/Dataset.html#show(numRows:Int,truncate:Int,vertical:Boolean):Unit) `.show()` method it limits the column width of the formatted output.

Also note [this](https://github.com/databrickslabs/overwatch/issues/1227#issuecomment-2135408925):
> I have overloaded `.showLines()` with the same set of method signatures as Spark's `Dataset.show()` for API consistency.